### ISSUE TITLE
GetDeviceWithFeaturesByRef fails on some HS3 devices ported to HS4

### ISSUE THREAD

**Issue by mads-b** [HomeSeer/Plugin-SDK#198]
Issue exhaustively described here:
https://forums.homeseer.com/forum/hs4-products/hs4-plugins/user-interface-plug-ins-aa/simmap-svamp/1472599-start-up-error-critical-error-on-initio-exception-while-exe

**Comment by brientim** [comment]
The error handling of the GetDeviceWithFeaturesByRef is the now problem and should exclude non-complaints feature and generate an exception report rather than stopping/failing.  

This error was previously reported and was supposedly fixed on a recent release. 

The features that caused the error in most causes did contain overlapping ranges of duplicate values but in the case of the z-wave sensor and the luminance feature I could not identify and offending overlapping range and this device was created by the recent updated z-wave plugin.

**Comment by jldubz** [comment]
@mads-b, can you provide the JSON describing an offending feature?
Thanks.

**Comment by spudwebb** [comment]
@jldubz  I can reproduce the problem with a Z-Wave device that has a luminance feature. This problem is now tracked as HS-1228 and PSDK-150

@mads-b  we are going to fix this problem, that being said:
-  you should try/catch GetDeviceWithFeaturesByRef  in your plugin so that if a device is badly configured and then GetDeviceWithFeaturesByRef throws an exception, it does not prevent your plugin from starting or processing other devices
- ideally you should not do anything too time-consuming in the Initialize method of your plugin. So if you are scanning and getting devices from HS in Initialize() you should probably create a new thread to do it instead.

**Comment by alexbk66** [comment]
> * you should probably create a new thread to do it instead

Or just (doesn't hurt even if you don't do much time-consuming)
```
UpdateConfigurationTask = Task.Factory.StartNew(() =>
{
     Update();
});
```

**Comment by mads-b** [comment]
Thanks for the feedback. I'll implement those improvements. That being said, and sorry for deviating from the issue:
Are there any docs on the SDK lifecycle of method invocations? You say I shouldn't freeze the thread calling Initialize, but why not? What is it doing that I shouldn't interrupt? Obviously it's not rendering UI.
Are there any way to find docs on what throws what? Visual studio provides no hints on HomeseerSystem methods and what they throw and why.

**Comment by spudwebb** [comment]
The issue with that specific  Z-Wave sensor device has been fixed in PluginSdk 1.3.0.0 and HS 4.1.17.0
